runDataExporter.sh

```diff
@@ -9,8 +9,18 @@ while (( "$#" )); do
 			build_jar=1
 			shift
 			;;
-		-c|--overwrite_config_file)
-			overwrite_config_file="--overwrite_config_file"
+		-c|--config_file)
+			if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
```

Comment:
What is `${2:0:1}` supposed to evaluate to?

Reply:
Honestly, I'm not sure - this was taken from https://medium.com/@Drew_Stokes/bash-argument-parsing-54f3b81a6a8f on parsing Bash arguments (documented on line 5 on this script)

Reply:
Ok, it might be worth trying to understand what's actually happening here.

Reply:
Okay, I'll try to understand it - it does do what we need, though.

Reply:
From what I found, ${2:0:1} means take the $2 variable (the value passed to the flag) and then parse it as a String starting at index 0 and take the first character.  So ${2:0:1} != "-" means the value passed with the flag doesn't start with a dash.